### PR TITLE
tab: init at 7.1

### PR DESCRIPTION
**Pull request by mstarzyk**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/tools/text/tab/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{ stdenv, fetchFromBitbucket, python2 }:
+
+stdenv.mkDerivation rec {
+  version = "7.1";
+  pname = "tab";
+  name = "${pname}-${version}";
```
Comment (alexarice) [inline pkgs/tools/text/tab/default.nix]: This is automatically set

File: pkgs/tools/text/tab/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  src = fetchFromBitbucket {
+    owner = "tkatchev";
+    repo = "${pname}";
```
Comment (alexarice) [inline pkgs/tools/text/tab/default.nix]:
```suggestion
    repo = pname;
```

Comment (alexarice) [inline pkgs/tools/text/tab/default.nix]:
```suggestion
    repo = pname;
```

File: pkgs/tools/text/tab/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    rev = "${version}";
```
Comment (alexarice) [inline pkgs/tools/text/tab/default.nix]:
```suggestion
    rev = version;
```

File: pkgs/tools/text/tab/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    sha256 = "049wx24b4v8alhvc05z7snyzaf6fk9w2npc4kklp366yjl1l4qbb";
+  };
+
+  buildInputs = [ python2 ];
```
Comment (peterhoeg) [inline pkgs/tools/text/tab/default.nix]: python2 is only used for running tests, so add this to nativeBuildInputs instead.

File: pkgs/tools/text/tab/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+  doCheck = true;
+
+  checkPhase = ''make test'';
```
Comment (peterhoeg) [inline pkgs/tools/text/tab/default.nix]: change to: ```checkTarget = "test";```

File: pkgs/tools/text/tab/default.nix [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+  installPhase = ''
```
Comment (peterhoeg) [inline pkgs/tools/text/tab/default.nix]: 1. missing "runHook preInstall" and postInstall
2. the 2 lines can be combined to ```install -Dm555 -t $out/bin tab```
3. there seems to be docs the source - please install those too

File: pkgs/tools/text/tab/default.nix [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    mkdir -p $out/bin
+    cp tab $out/bin
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Programming language/shell calculator";
+    homepage    = https://tkatchev.bitbucket.io/tab/;
```
Comment (peterhoeg) [inline pkgs/tools/text/tab/default.nix]: quote the url

**Review by peterhoeg — CHANGES_REQUESTED** [review]

File: pkgs/tools/text/tab/default.nix (hunk 1 above)
Comment (jonringer) [inline pkgs/tools/text/tab/default.nix]: unneeded
```suggestion
```

File: pkgs/tools/text/tab/default.nix (hunk 6 above)
Comment (mstarzyk) [reply pkgs/tools/text/tab/default.nix]: 1. Added hooks. Thanks.
2. Yes, 2 lines can be combined,  what would be an advantage of that solution, besides saving one line :) ?
3. Yes, there are docs, but "make install" does not install them, and also [installation instruciton](https://tkatchev.bitbucket.io/tab/docs.html#compiling-and-installing) says to just copy the excutable. Man page would be useful, but there isn't one. There is quite extensive help from the excutable though (`tab -h`).

File: pkgs/tools/text/tab/default.nix (hunk 7 above)
Comment (mstarzyk) [reply pkgs/tools/text/tab/default.nix]: Why do you recommend quoting? Most examples at [Nixpkgs Contributors Guide](https://nixos.org/nixpkgs/manual/) are unquoted.

Comment (peterhoeg) [reply pkgs/tools/text/tab/default.nix]: https://github.com/NixOS/rfcs/pull/45

File: pkgs/tools/text/tab/default.nix (hunk 6 above)
Comment (peterhoeg) [reply pkgs/tools/text/tab/default.nix]: Regd docs, please copy them in. It's nicer for the end user if there is something more than just a binary regardless of what upstream installs.

Comment (peterhoeg) [reply pkgs/tools/text/tab/default.nix]: > Yes, 2 lines can be combined, what would be an advantage of that solution, besides saving one line :) ?

```install``` allows you to create the directory, copy files, set permissions and set ownership all in one go which makes it much easier to read in general (not just in the context of nix) as it keeps everything that touches a certain file or set of files together in one place as opposed to spread out.

**Comment by mstarzyk** [comment]
Version 7.2 is available, so I am closing this PR, and submitting a new one: #71466

**Pull request closed without merging**